wakuv2/filter_manager.go

```diff
@@ -210,16 +207,16 @@ func (mgr *FilterManager) subscribeToFilter(filterID string, peer peer.ID, tempI
 		return
 	}
 	contentFilter := mgr.buildContentFilter(f.PubsubTopic, f.ContentTopics)
-	logger.Debug("filter subscribe to filter node", zap.Stringer("peer", peer), zap.String("pubsubTopic", contentFilter.PubsubTopic), zap.Strings("contentTopics", contentFilter.ContentTopicsList()))
+	logger.Debug("filter subscribe to filter node", zap.String("pubsubTopic", contentFilter.PubsubTopic), zap.Strings("contentTopics", contentFilter.ContentTopicsList()))
 	ctx, cancel := context.WithTimeout(mgr.ctx, requestTimeout)
 	defer cancel()
 
-	subDetails, err := mgr.node.FilterLightnode().Subscribe(ctx, contentFilter, filter.WithPeer(peer))
+	subDetails, err := mgr.node.FilterLightnode().Subscribe(ctx, contentFilter, filter.WithAutomaticPeerSelection(mgr.peers...))
 	var sub *subscription.SubscriptionDetails
 	if err != nil {
-		logger.Warn("filter could not add wakuv2 filter for peer", zap.Stringer("peer", peer), zap.Error(err))
+		logger.Warn("filter could not add wakuv2 filter for peers", zap.Any("peers", mgr.peers), zap.Error(err))
 	} else {
-		logger.Debug("filter subscription success", zap.Stringer("peer", peer), zap.String("pubsubTopic", contentFilter.PubsubTopic), zap.Strings("contentTopics", contentFilter.ContentTopicsList()))
+		logger.Debug("filter subscription success", zap.Any("peers", mgr.peers), zap.String("pubsubTopic", contentFilter.PubsubTopic), zap.Strings("contentTopics", contentFilter.ContentTopicsList()))
```

Comment:
Rather than mgr.peers, better to print the actual peerID with which filter subscription was successful. You can get that from `subDetails.PeerID`

Reply:
Also where is mgr.peers getting populated? We need to see if that is being used somewhere else and if it is being directly taken from node peer-store then it maybe inconsistent with actual peers with which filter subs are maintained. 

I think we may have to do detailed analysis of how filter state is maintained in status-go and update things accordingly. I am not sure changing peer selection to automatic would just be this much.

PingPeers is done based on peerIDs that are part of subscriptions, which is fine and not impacted by this change. 

But unsubscribe and any othre places where mgr.Peers is used need to taken a look at., because mgr.filterPeers uses peerStore to find filterPeers. It is possible that there maybe filterPeers in the peerStore but not subscribed to. We need to ensure Filter operations like unsubscribe etc are not sent to peers to which subscriptions are not present.

I am not sure if there is some other peer reference that needs to be looked at.

Reply:
Chiming in here - unsubscribes are performed via `UnsubscribeWithSubscription`, this should ensure that we're using a particular peer that already is being subscribed to.
`FilterManager.peers` is populated on a scheduled basis, using libp2p's PeerStore. If some peer pings fail in the time interval between full peer re-population, these disconnected peers are removed from `FilterManager.peers` (possibly to be added back during full re-population)

Reply:
> Chiming in here - unsubscribes are performed via `UnsubscribeWithSubscription`, this should ensure that we're using a particular peer that already is being subscribed to.

In that case we don't have to worry about unsubscribes.

Reply:
> `FilterManager.peers` is populated on a scheduled basis, using libp2p's PeerStore. If some peer pings fail in the time interval between full peer re-population, these disconnected peers are removed from `FilterManager.peers` (possibly to be added back during full re-population)

If we are relying on peer-selection from go-waku, then I don't think this is required. Rather the peers selected for each filter can be stored in mgr.peers and pings can be sent to them accordingly.

Reply:
Agree with this option - with automatic peer selection we don't need to refer to peer store. In that case we don't even need `mgr.peers` i think - it ceases to be a pool of possible peer candidates and instead becomes a duplicate store of peer IDs, given that we can fetch all current peer IDs from active subscriptions anyway.

Reply:
Kind of agree to it, but how will Filter ping be issued. AFAIU, right now it is being issued based on peers listed in mgr.peers. 
Otherwise, the subscriptions can be looped and aggregated peer-wise to issue the pings.